integration_tests/src/main/python/parquet_test.py

```diff
@@ -199,7 +199,7 @@ def test_ts_read_fails_datetime_legacy(gen, spark_tmp_path, ts_write, ts_rebase,
             lambda spark : readParquetCatchException(spark, data_path),
             conf=all_confs)
 
-@pytest.mark.parametrize('parquet_gens', [decimal_gens,
+@pytest.mark.parametrize('parquet_gens', [[byte_gen, short_gen], decimal_gens,
```

Comment:
Why add `byte_gen` and `short_gen` as their own specific test cast to a test that is only about decimals?

Reply:
The test array is missing DecimalGen, it should be [byte_gen, short_gen, DecimalGen(7, 3)]. There was a bug in the old code which would pop the precisionList even if the element wasn't a Decimal